### DISCUSSION TITLE
Fabric Coordinates

### DISCUSSION THREAD

**Discussion by ralfHielscher** [mtex-toolbox/mtex#872] [Feature Requests]
Implement a new class representing a fabric coordinate system similar to crystalSymmetry representing a crystal coordinate system such that we can have an analog to

* Miller but representing fabric directions
* orientation but representing fabric orientations
* ODF but representing the distribution of fabric orientations

Internal representation: what do we need to describe the fabric system:
* foliation direction?
* foliation plane?

How do we represent a fabric orientation?
* dipdir / strike, dip, trend/ plunge or
* rake / pitch

Conventions: mapping x,y,z to N,E,H

 Q: what to do with 4 parameters and treat non-coplanarity ("the line is not exactly in the plane")

  b) fabric "Millers": foliation, lineation, (maybe section normals ?),others

  c) mmm or monoclinic or triclinic
     Q: how to treat rotations larger 360 degree (rare but possible)

  d) functions: rotate2geo, mode positions of maxima etc...

ori = orientation.byTrendPlunge()
ori = orientation.by()

**Comment by kilir** [comment]
Update:
implementing fabric-lineation defined orientation

@geoOrientation (foliation/lineation object) < geoOrientation

input either
by two vector3d
    a) lineation (vector3d.byTrendPlunge)
    b) foliation (vector3d.byDipdirDip or vector3d.byStrikeDip)
    
-> needs some threshold since L might not be eaxctly in the plane (due to measurement)
   resp. just take the lineation and project it into the foliation
-> foliation: what is measured is the plane
    dipdir+dip (=direction of higest gradient and inclinatin against horizontal) 
    or 
    strike and dip (= direction of 0-gradient and inclination against horizontal, right hand rule)
    but of course be presented by the plane normal
    
or by 3 angles
    a) dipdirection, dip, rake(pitch)
    b) strike, dip, rake/pitch


Q1: should @vector3d get vector3d.byTrendPlunge, vector3d.byDipdirDip and vector3d.byStrikeDip
    or is this too much and do something vector3d.byGeo(...,...,'dipdirdip') covering all sorts of situations
    
Q2: What symmetry should @geoOrientation have?
    ... and how to assign (automatic, user defined...)
    (this makes a difference when computing angles between two orientations, when they are far away)
    
a) 2 situations: folition is polar(f_p , e.g. old/young) or antipodal(f_a)
b) 2 situations: lineation is directional (l_p, e.g. transport direction / "shear sense")  or axial/antipodal (l_a)
    
  combinations and possible symmetry elements:
 ```   
  f_a   l_a       '222'
  f_p   l_a        '2' || foliation normal
    
  f_p   l_p       '1' 
  f_a   l_p       '2' || cross(foliation, lineation)
  ```  
  I might be wrong with the latter suggestion.
    
Q3: What is a more reasonable name?
